.github/workflows/ci.yml

```diff
@@ -28,7 +28,32 @@ jobs:
           idf.py build
 
       - name: Archive release
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
           name: elf
           path: build/*.elf.memfault_log_fmt
+
+  build-with-component:
+    runs-on: ubuntu-latest
+    container:
+      image: espressif/idf:release-v5.3
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Build
+        run: |
+          # set a phony project key
+          echo "CONFIG_MEMFAULT_PROJECT_KEY=\"1234\"" >> sdkconfig.defaults
+          . ${IDF_PATH}/export.sh
+
+          # add the Memfaut component
+          idf.py add-dependency "memfault/memfault-firmware-sdk^1.15.0"
```

Comment:
I wonder if there is a better way to sync this version with the one in the directory's submodule. Could we read in the `VERSION` file? To steal your method from port-playground:

```
git submodule update --init
SDK_VERSION=$(cat third-party/memfault-firmware/sdk/VERSION | grep -oP 'VERSION:\s*\K[^[:space:]]+')
```

Reply:
good idea! will do